File: api/Core/Model/CustomerAddress.cs
using System;

namespace Xcianify.Core.Model
{
    public class CustomerAddress : BaseModel
    {
        public int CustomerId { get; set; }
        public string? AddressLine1 { get; set; }
        public string? AddressLine2 { get; set; }
        public string? AddressLine3 { get; set; }
        public string? City { get; set; }
        public string? ZipCode { get; set; }
        public string? Country { get; set; }
        public string? StateCode { get; set; }
        public string? GstStateCode { get; set; }
        public string? ContactPerson { get; set; }
        public string? TelephoneNumber { get; set; }
        public string? MobileNumber { get; set; }
        public string? FaxNumber { get; set; }
        public string? EmailId { get; set; }
        public string? Website { get; set; }
        public bool IsPrimary { get; set; } = true;
    }
} 

File: pharma-erp-3/types/customer-address.ts
export interface CustomerAddress {
  id: number;
  customerId: number;
  addressLine1?: string;
  addressLine2?: string;
  addressLine3?: string;
  city?: string;
  zipCode?: string;
  country?: string;
  stateCode?: string;
  gstStateCode?: string;
  contactPerson?: string;
  telephoneNumber?: string;
  mobileNumber?: string;
  faxNumber?: string;
  emailId?: string;
  website?: string;
  isPrimary: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CustomerAddressFilter {
  customerId: number;
  isPrimary?: boolean;
  page: number;
  pageSize: number;
  sortBy: string;
  sortDescending: boolean;
}

export type CreateCustomerAddressData = Omit<CustomerAddress, 'id' | 'createdAt' | 'updatedAt'>;

export type UpdateCustomerAddressData = Partial<Omit<CustomerAddress, 'id' | 'createdAt' | 'updatedAt'>> & {
  id: number;
};
